packages/grafana-ui/src/components/Modal/Modal.tsx

```diff
@@ -19,48 +19,51 @@ export interface Props extends Themeable {
 
   /** If not set will call onDismiss if that is set. */
   onClickBackdrop?: () => void;
+  render?: (props: RenderApi) => JSX.Element;
 }
 
-export class UnthemedModal extends React.PureComponent<Props> {
-  onDismiss = () => {
-    if (this.props.onDismiss) {
-      this.props.onDismiss();
-    }
-  };
-
-  onClickBackdrop = () => {
-    this.onDismiss();
-  };
+interface RenderApi {
+  styles: ReturnType<typeof getModalStyles>;
+}
 
-  renderDefaultHeader(title: string) {
-    const { icon, iconTooltip } = this.props;
+export function Modal(props: PropsWithChildren<Props>): JSX.Element | null {
```

Comment:
I'm using `function` not `const` here so I can't use FC? Any suggestions?

Reply:
I updated the return type to `ReturnType<FC<Props>>` instead.

Reply:
Any particular reason we'd want to use `function` here? I think in this case const + FC make code a bit cleaner. Not sure there's a way to use FC with `function` keyword.

Reply:
We should have a Frontend Weekly discussion about this. Seems like we don't really have a clear guideline regarding this (yet). Until we have I'll keep using `function` 😉

Reply:
My two cents about this. For a matter of readability, I would opt for:

- `function` - when a function actually is a utility
- `const` - in case of _function component_. It's easier to type with `:FC<..>` than `ReturnType<FC<...>>` and I guess I've  never seen typing function components this way. Guess event React guide uses const all over the place rather than functions.

Reply:
Hahaha, I love being the first one using that instead of `ReactElement | null`. We should really have a Frontend discussion about the `function` vs `const` usage. I know that @torkelo has some opinions about this too.

Reply:
yes, I prefer function over a const variable, except for react components, where the typing is easier / more readable with a const variable.

Reply:
Ok, I didn't realize you preferred const for ReactComponents. Good to know.